File: ReverseVowelsOfaString/ReverseVowels.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function (s) {
  let vowels = ["a", "A", "e", "E", "i", "I", "o", "O", "u", "U"];
  let stack = [];
  for (let i = 0; i < s.length; i++) {
    if (vowels.indexOf(s[i]) !== -1) {
      stack.push(s[i]);
    }
  }
  let ans = [];
  for (let i = 0; i < s.length; i++) {
    if (vowels.indexOf(s[i]) !== -1) {
      ans.push(stack.pop());
    } else {
      ans.push(s[i]);
    }
  }
  return ans.join("");
};

File: ReverseVowelsOfaString/ReverseVowels.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = set("aeiouAEIOU")
        s = list(s)
        pointer1 = 0
        pointer2 = len(s) - 1
        while pointer1 < pointer2:
            if s[pointer1] in vowels and s[pointer2] in vowels:
                s[pointer1], s[pointer2] = s[pointer2], s[pointer1]
            elif s[pointer1] not in vowels:
                pointer1 += 1
                continue
            elif s[pointer2] not in vowels:
                pointer2 -= 1
                continue
            pointer1 += 1
            pointer2 -= 1
        return ''.join(s)